### PR TITLE
[TS-244] Added activities route

### PR DESCRIPTION
**Pull request by tungleu**
added endpoint to get recent activities 
`GET /actitivities/:userId`
image

### CODE REVIEW

File: src/controllers/activities.controller.ts
Comment (bowe99) [inline src/controllers/activities.controller.ts:17]: This should be renamed to activities

File: src/controllers/activities.controller.ts
Comment (bowe99) [inline src/controllers/activities.controller.ts:31]: This code is failing Sonar because of duplication maybe you can clean it up a little

**Review by bowe99 — CHANGES_REQUESTED** [review]

File: src/routes/v1/activities.route.ts
Comment (bowe99) [inline src/routes/v1/activities.route.ts:3]: This is not being used you can remove it

**Review by bowe99 — CHANGES_REQUESTED** [review]

File: src/db/service/alpaca.service.ts
Comment (kennguyen0303) [inline src/db/service/alpaca.service.ts:20]:
```suggestion
			const {data}= await axios.get(endPoint, {
				headers: { Authorization: `Bearer ` + alpacaToken },
			});
			return res.send(messages.successMessage("success", infoType, data));
```

Save you one line and one variable

File: src/controllers/positions.controller.ts
Comment (kennguyen0303) [inline src/controllers/positions.controller.ts:20]:
```suggestion
			return await this.alpacaService.getInfo(req, res, "/positions", "positions", userInfo.alpacaToken);
```

missing `await`

File: src/controllers/positions.controller.ts
Comment (kennguyen0303) [inline src/controllers/positions.controller.ts:22]:
```suggestion
			return res.send(messages.internalError("User hasn't linked any Alpaca account"));
```

Missing `return` for convention

File: src/controllers/activities.controller.ts
Comment (kennguyen0303) [inline src/controllers/activities.controller.ts:19]:
```suggestion
			return await this.alpacaService.getInfo(
```

missing `await`

File: src/controllers/account.controller.ts
Comment (kennguyen0303) [inline src/controllers/account.controller.ts:22]:
```suggestion
			return res.send(messages.internalError("User hasn't linked any Alpaca account"));
```

missing `return`

File: src/controllers/account.controller.ts
Comment (kennguyen0303) [inline src/controllers/account.controller.ts:20]:
```suggestion
			return await this.alpacaService.getInfo(req, res, "/account", "account", userInfo.alpacaToken);
```

missing `await`

**Review by kennguyen0303 — CHANGES_REQUESTED** [review]
Great work, only 2 small things you might have known. I think having `await` and `return` is necessary. Although there might be no issue on your method, it triggers unexpected behavior sometimes if you work on a different feature. 

Food for thoughts: `await` will tell NodeJS to mark that line, run that function in synchronous behavior and wait for a return; As a result, if your workflow contains only one method, it won't make any difference. Without `await`, nodejs will jump over your call without waiting for the return/response, possibly go to the next lines of code (if there any).

Thanks @bowe99 for noticing the unused imports, I would have missed them for sure.

**Review by kennguyen0303 — COMMENTED** [review]
I would like to have a unit test for the service before this is merged. Will it be possible?

**Review by kennguyen0303 — DISMISSED** [review]
lgtm!

**Review by kennguyen0303 — APPROVED** [review]

**Comment by kennguyen0303** [comment]
@bowe99 need your review here since you requested some changes.

**Review by kennguyen0303 — APPROVED** [review]

**Review by GadoliniumPotassium — APPROVED** [review]

**Pull request merged**
